engine/consensus/dkg/reactor_engine.go

```diff
@@ -79,7 +82,38 @@ func NewReactorEngine(
 // that will close when the engine has successfully
 // started.
 func (e *ReactorEngine) Ready() <-chan struct{} {
-	return e.unit.Ready()
+	return e.unit.Ready(func() {
+		// If we are starting up in the EpochSetup phase, try to start the DKG.
+		// If the DKG for this epoch has been started previously, we will exit
+		// and fail this epoch's DKG.
```

Comment:
Either of those 2 cases are possible here. 

1. We started the DKG previously and restarted before completing it. 
We don't have a way to safely recover an in-progress DKG currently, so the only safe course of action here is to exit the DKG process and consider this a locally failed DKG.

2. We started the DKG previously and completed it, then restarted.
This is an uncommon case, because there is typically very little time between local completion of the DKG and the end of the `EpochSetup` phase. But in this case we want to take the same action as Case 1 - exit the DKG process, since we already completed it.